front/src/modules/trainschedule/components/Timetable/TimetableTrainCard.tsx

```diff
@@ -269,7 +269,7 @@ const TimetableTrainCard = ({
             </div>
             <div className="duration-time">
               <span data-testid="train-duration">
-                {dayjs.duration(train.duration).format('HH[h]mm')}
+                {dayjs.duration(train.duration?.ms ?? 0).format('HH[h]mm')}
```

Comment:
why don’t you keep 0 as default value in `formatTrainScheduleSummaries.ts`
 if you end up using 0 anyway if it’s null?

Reply:
makes sense

i guess you can go for `train.duration!.ms`. But this assumes that we enforce long term that we only ever create valid train with non-null duration.

Reply:
Done!

> this assumes that we enforce long term that we only ever create valid train with non-null duration.

This should always be the case: if the train is valid we can simulate it, if we can simulate it then we have an arrival time and total duration. BTW, `arrivalTime` is already nullable.